src/MigrationTools/DataContracts/Pipelines/TaskGroups.cs

```diff
@@ -8,130 +8,93 @@ namespace MigrationTools.DataContracts.Pipelines
-        public string Id { get; set; }
-
-        [JsonProperty("name")]
-        public string Name { get; set; }
-
-        [JsonProperty("version")]
         public Version Version { get; set; }
 
-        [JsonProperty("iconUrl")]
         public string IconUrl { get; set; }
 
-        [JsonProperty("friendlyName")]
         public string FriendlyName { get; set; }
 
-        [JsonProperty("description")]
         public string Description { get; set; }
 
-        [JsonProperty("category")]
         public string Category { get; set; }
 
-        [JsonProperty("definitionType")]
         public string DefinitionType { get; set; }
 
-        [JsonProperty("author")]
         public string Author { get; set; }
 
-        [JsonProperty("demands")]
         public object[] Demands { get; set; }
 
-        [JsonProperty("groups")]
         public object[] Groups { get; set; }
 
-        [JsonProperty("inputs")]
         public Input[] Inputs { get; set; }
 
-        [JsonProperty("satisfies")]
         public object[] Satisfies { get; set; }
 
-        [JsonProperty("sourceDefinitions")]
         public object[] SourceDefinitions { get; set; }
 
-        [JsonProperty("dataSourceBindings")]
         public object[] DataSourceBindings { get; set; }
 
-        [JsonProperty("instanceNameFormat")]
         public string InstanceNameFormat { get; set; }
 
-        [JsonProperty("preJobExecution")]
         public Execution PreJobExecution { get; set; }
 
-        [JsonProperty("execution")]
         public Execution Execution { get; set; }
 
-        [JsonProperty("postJobExecution")]
         public Execution PostJobExecution { get; set; }
 
-        [JsonProperty("comment", NullValueHandling = NullValueHandling.Ignore)]
         public string Comment { get; set; }
 
-        [JsonProperty("visibility", NullValueHandling = NullValueHandling.Ignore)]
         public string[] Visibility { get; set; }
 
-        [JsonProperty("owner", NullValueHandling = NullValueHandling.Ignore)]
         public string? Owner { get; set; }
 
-        [JsonProperty("contentsUploaded", NullValueHandling = NullValueHandling.Ignore)]
         public bool? ContentsUploaded { get; set; }
 
-        [JsonProperty("packageType", NullValueHandling = NullValueHandling.Ignore)]
         public string PackageType { get; set; }
 
-        [JsonProperty("packageLocation", NullValueHandling = NullValueHandling.Ignore)]
         public string PackageLocation { get; set; }
 
-        [JsonProperty("sourceLocation", NullValueHandling = NullValueHandling.Ignore)]
         public string SourceLocation { get; set; }
 
-        [JsonProperty("minimumAgentVersion", NullValueHandling = NullValueHandling.Ignore)]
         public string MinimumAgentVersion { get; set; }
 
-        [JsonProperty("helpMarkDown", NullValueHandling = NullValueHandling.Ignore)]
         public string HelpMarkDown { get; set; }
+
+        public override RestApiDefinition GetMigrationObject()
```

Comment:
I would call this method something like Reset or similar, now the name does not imply what it is actually doing, aka reset values that cannot be set on new objects?